app/services/search/buffer_suggestions_builder.rb

```diff
@@ -0,0 +1,22 @@
+class Search::BufferSuggestionsBuilder
+  include DistanceHelper
+
+  attr_reader :location, :buffer_suggestions, :search_params
+
+  def initialize(location, search_params)
+    @location = location
+    @search_params = search_params
+    get_buffers_suggestions
+  end
+
+  def get_buffers_suggestions
+    buffer_counts = LocationPolygon.with_name(location).buffers.each_with_object([]) do |buffer, buffer_counts|
```

Comment:
This would look cleaner imo if you just mapped `LocationPolygon.with_name(location).buffers`